eng/pipelines/docs.yml

```diff
@@ -3,79 +3,98 @@ trigger:
+          script: |
+            mdoc.exe update -fx ../dll-docs -o ../dll-xml-output --debug -lang docid -lang vb.net -lang fsharp --delete
+          workingDirectory: $(Build.BinariesDirectory)/mdoc
           failOnStderr: true
-      - task: PythonScript@0
-        displayName: "Arrange Yaml files into Folders"
+      - task: CmdLine@2
+        displayName: Generate YAML
         inputs:
-          scriptPath: "$(Build.SourcesDirectory)/eng/docgeneration/script/MoveYmlFiles.py"
-          arguments: $(Build.SourcesDirectory)
+          script: |
+            "$(Build.BinariesDirectory)/ECMA2Yml/ECMA2Yaml.exe" -s "./dll-xml-output" -o "./dll-yaml-output"
+          workingDirectory: $(Build.BinariesDirectory)
           failOnStderr: true
-      - task: PythonScript@0
-        displayName: "Create Toc.yml for each service directory"
+      - task: CmdLine@2
+        displayName: Provision DocFX Directory
         inputs:
-          scriptPath: "$(Build.SourcesDirectory)/eng/docgeneration/script/CreateTocYml.py"
-          arguments: $(Build.SourcesDirectory)
+          script: $(Build.BinariesDirectory)/docfx/docfx.exe init -q
+          workingDirectory: $(Build.BinariesDirectory)/docfx-output
+          failOnStderr: true
       - pwsh: |
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/api/*" -Destination "$(Build.SourcesDirectory)/docfx_project/api/" -Force
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/api/index.md" -Destination "$(Build.SourcesDirectory)/docfx_project/articles/intro.md" -Force
-          Copy-Item "$(Build.SourcesDirectory)/Documentation/*" -Destination "$(Build.SourcesDirectory)/docfx_project/articles/"
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/index/*" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
-          Copy-Item "$(Build.SourcesDirectory)/CONTRIBUTING.md" -Destination "$(Build.SourcesDirectory)/docfx_project/articles/"
-          Copy-Item "$(Build.SourcesDirectory)/packages.md" -Destination "$(Build.SourcesDirectory)/docfx_project/articles/"
-          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/docfx.json" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
+          Copy-Item "$(Build.BinariesDirectory)/dll-yaml-output/*" -Destination "$(Build.BinariesDirectory)/docfx-output/docfx_project/api/" -Recurse
+          Copy-Item "$(Build.SourcesDirectory)/README.md" -Destination "$(Build.BinariesDirectory)/docfx-output/docfx_project/index.md" -Recurse -Force
+          Copy-Item "$(Build.SourcesDirectory)/README.md" -Destination "$(Build.BinariesDirectory)/docfx-output/docfx_project/api/index.md" -Recurse -Force
+          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/assets/docfx.json" -Destination "$(Build.BinariesDirectory)/docfx-output/docfx_project" -Recurse -Force
+          Remove-Item -Path "$(Build.BinariesDirectory)/docfx-output/docfx_project/toc.yml" -Force
```

Comment:
Is there something else that updates toc.yml? If not why do we need to keep overwriting it? Also is there a reason we don't just check this file in and copy the entire file instead of writing the same content each time?

Reply:
The default toc.yml theat DocFx will generate is not what we want hence the overwrite. I was actually trying to avoid checking another file into the repo. Do you want me to check it in?

Reply:
Given it is static data I would check it in as it will be easier to understand. Writing to the file like this makes me think that we dynamically generating this and it just introduces more points of potential failure.